src/sharing/ShareWrapper.js

```diff
@@ -177,19 +177,22 @@ class ShareWrapperInner extends React.Component<Props, State> {
     } catch (err) {
       showToast(_('Failed to send message'));
       logging.error(err);
-      this.onShareCancelled();
+      this.onShareCancelled(sendTo);
       return;
     }
     showToast(_('Message sent'));
-    this.onShareSuccess();
+    this.onShareSuccess(sendTo);
   };
 
-  onShareCancelled = () => {
+  onShareCancelled = sendTo => {
+    // If in the future this callback uses the `sendTo` data, it should get
+    // it from its parameter (just like `onShareSuccess` does) and not from
+    // the props.  That's because the props may have changed since the
+    // actual send request we just made.
```

Comment:
Yeah, disallowing that in the UI would be quite reasonable.

Even then, I think it'd be cleaner for these callbacks to get `sendTo` from their caller (as they do with this change), rather than from the component's props (as they do before). That way the fact that this code all agrees on what conversation it's referring to is something that's straightforward to see locally -- the information gets read once and then passed around -- rather than have to wonder if there might be a way for it to get mutated between the different times we read it off the props.